src/core_modules/capture-core/components/WidgetEnrollment/Actions/Actions.component.js

```diff
@@ -32,17 +32,29 @@ export const ActionsPlain = ({
     loading,
     onlyEnrollOnce,
     classes,
-}: PlainProps) => (
-    <>
-        <DropdownButton
-            dataTest="widget-enrollment-actions-button"
-            secondary
-            small
-            disabled={loading}
-            className={classes.actions}
-            component={
-                <span>
-                    {loading ? null : (
+}: PlainProps) => {
+    const [open, setOpen] = useState(false);
+    const handleOnUpdate = (arg) => {
+        setOpen(prev => !prev);
+        onUpdate(arg);
+    };
+    const handleOnDelete = (arg) => {
+        setOpen(prev => !prev);
```

Comment:
since delete action will navigate to dashboard page, do we need to close the menu?

Reply:
Hey,
Yes, the menu needs to be closed to handle the case when the Delete enrollment request fails. In this scenario, the user stays on the same page and the menu should be closed.
Thanks!